meta/saiserialize.c

```diff
@@ -576,10 +576,38 @@ int sai_deserialize_macsec_auth_key(
         return read;
     }
 
-    SAI_META_LOG_WARN("failed to deserialize '%.*s' as macsec_auth_key", MAX_CHARS_PRINT, buffer);
+    SAI_META_LOG_WARN("failed to deserialize '%.*s' as auth_key", MAX_CHARS_PRINT, buffer);
     return SAI_SERIALIZE_ERROR;
 }
 
+int sai_serialize_macsec_sak(
+        _Out_ char *buffer,
+        _In_ const sai_macsec_sak_t sak)
+{
+   return sai_serialize_encrypt_key( buffer, sak );
+}
+
+int sai_deserialize_macsec_sak(
+        _In_ const char *buffer,
+        _Out_ sai_macsec_sak_t sak)
+{
+   return sai_deserialize_encrypt_key( buffer, sak );
+}
+
+int sai_serialize_macsec_auth_key(
+        _Out_ char *buffer,
+        _In_ const sai_macsec_auth_key_t auth)
+{
+   return sai_serialize_auth_key( buffer, auth );
+}
+
+int sai_deserialize_macsec_auth_key(
+        _In_ const char *buffer,
+        _Out_ sai_macsec_auth_key_t auth)
+{
+   return sai_deserialize_auth_key( buffer, auth );
```

Comment:
In the sai-ipsec subgroup meeting on July 14, it was decided to duplicate the data type for IPsec because of the unfortunate naming of the MACsec data type.
The existing sai_macsec_auth_key_t and the corresponding enum SAI_MACSEC_SA_ATTR_AUTH_KEY should not be changed because those are used in existing MACsec drivers and NOS code.
Using sai_macsec_auth_key_t for IPsec could have worked with an enum SAI_IPSEC_SA_ATTR_MACSEC_AUTH_KEY, but the MACsec reference in IPsec context is weird and confusing.  Hence duplication was preferred.

Reply:
instead of having 2 different types, can we have 1 type (new one) and add a "define OLD_type NEW_type" ??

Reply:
> In the sai-ipsec subgroup meeting on July 14, it was decided to duplicate the data type for IPsec because of the unfortunate naming of the MACsec data type.

Why SAI_MACSEC_ATTR_MACSEC_AUTH_KEY is needed when MACsec Secure Association created?
MACsec is based on 802.1AE cipher suite. 
Data plane encryption is based on GCM-AES-128/256 or GCM-XPN-AES-128/256.
Data plane authentication is based on GHASH function as GCM Spec.
What can the MACsec crypto hardware do with an SAI_MACSEC_ATTR_MACSEC_AUTH_KEY ?